pkgs/development/libraries/tinycdb/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "tinycdb-${version}";
+  version = "0.78";
+  outputs = [ "out" "dev" "lib" "man" ];
+  separateDebugInfo = true;
+  postPatch = ''
+    substituteInPlace Makefile --replace 'prefix=/usr/local' 'prefix=$(out)'
+  '';
+  buildPhase = ''
+    make staticlib sharedlib cdb-shared
+  '';
+  installPhase = ''
+    make install
+    mkdir -p $lib/lib $dev/lib $out/bin
+    cp libcdb.so* $lib/lib
+    cp cdb-shared $out/bin/cdb
+    mv $out/lib/libcdb.a $dev/lib
+    rmdir $out/lib
+  '';
+
+  src = fetchurl {
+    url = "http://www.corpit.ru/mjt/tinycdb/${name}.tar.gz";
+    sha256 = "0g6n1rr3lvyqc85g6z44lw9ih58f2k1i3v18yxlqvnla5m1qyrsh";
+  };
+
+  meta = with lib; {
+
+    description = "utility to manipulate constant databases (cdb)";
+    outputsToInstall = outputs;
```

Comment:
Is there any reason to install the dev and lib outputs ?

Reply:
yes but you are not supposed to install libraries with nix: https://nixos.wiki/wiki/FAQ/I_installed_a_library_but_my_compiler_is_not_finding_it._Why%3F

Reply:
True. But I do not want application (e.g tinycdb.out) to have cdb.h in its closure

Reply:
I think @symphorien is asking why you are installing the dev output by default? He linked to a wiki page which says not to do that.